### ISSUE TITLE
SwiftFormat breaks build for one liner that combines `try` and  `??`

### ISSUE THREAD

**Issue by ncooke3** [nicklockwood/SwiftFormat#1142]
Using latest **SwiftFormat 0.49.4** version.

SwiftFormat accidentally breaks builds when trying to style
a statement that combines `try` and  `??`.

Consider the below snippet:
```swift
func foo(_ bar: String?) throws -> String {
    return try bar ?? { throw CustomError.emptyValue }()
}
```

The returning one-liner is a bit funky, but is supposed to read like: 
> return this optional string's value if it is non-nil; else, throw an error

Formatting the above swift snippet results in the following diff.
```diff
-     return try bar ?? { throw CustomError.emptyValue }()
+     return try bar ?? throw CustomError.emptyValue          // 🔺 Expected expression after operator
```

The new diff breaks fails to compile with the following error:
```
main.swift:11:23: error: expected expression after operator
    return try bar ?? throw CustomError.emptyValue
```
---
I have a repo with instructions here: https://github.com/ncooke3/repros/tree/main/swiftformat-1142

**Comment by nicklockwood** [comment]
@ncooke3 thanks for reporting this. Disabling the `redundantClosure` rule should fix it for now.

**Comment by nicklockwood** [comment]
cc: @calda

**Comment by calda** [comment]
Interesting! This rule probably need to filter out closures that include `throw`. Should be a quick fix, will take a look.

**Comment by nicklockwood** [comment]
@calda don't worry, I have a fix already.

**Comment by calda** [comment]
Great, thanks!

**Comment by nicklockwood** [comment]
@ncooke3 fixed in 0.49.5
